django/contrib/postgres/fields/array.py

```diff
@@ -288,3 +289,13 @@ def __init__(self, start, end):
 
     def __call__(self, *args, **kwargs):
         return SliceTransform(self.start, self.end, *args, **kwargs)
+
+
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, None)
+@CombinedExpression.dispatch(CombinedExpression.ADD, None, ArrayField)
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, ArrayField)
```

Comment:
```
Ambiguities exist in dispatched function +

The following signatures may result in ambiguous behavior:
	[CharField, Field], [Field, ArrayField]
	[ArrayField, Field], [Field, CharField]


Consider making the following additions:

@dispatch(CharField, ArrayField)
def +(...)

@dispatch(ArrayField, CharField)
def +(...)
```
This is raised from `multipledispatch` on registration (!), not during query execution. I think it's a Good Thing (TM).